### ISSUE TITLE
\n in an exception

### ISSUE THREAD

**Issue by mrakgr** [fsprojects/FsUnit#96]
Program:
```
open FsUnitTyped
open NUnit.Framework

1 |> shouldEqual 2
1 |> shouldNotEqual 2
```
Output:
```
NUnit.Framework.AssertionException:   Expected: 2\nActual: 1
  Expected: True
  But was:  False
```
What bothers me is that `\n` that is printed as a part of the actual string. This is my first time using both FsUnit and NUnit, but that can't be right.

**Comment by sergey-tihon** [comment]
I guess it is a new side effect of NUnit 3

What we do in the code now:
```fsharp
    [<DebuggerStepThrough>]
    let shouldEqual (expected : 'a) (actual : 'a) =
        Assert.IsTrue((expected = actual), sprintf "Expected: %A\nActual: %A" expected actual)
```
We need `Assert.IsTrue` to use F# structural equality here.

@mrakgr Would you like to find a way to pass message string correctly to NUnit 3?

//cc: @rprouse @et1975

**Comment by mrakgr** [comment]
Hmmm...I took a look at [Don's blog](https://blogs.msdn.microsoft.com/dsyme/2009/11/08/equality-and-comparison-constraints-in-f/) and it seems structural equality constraints are set directly on the classes. I was also under the impression that structural equality is the default in F# for the relevant types. Does the Assert class need to be modified to use structural equality then? That an NUnit class written in C# right?

As for your question, right now, I am still getting familiar with testing frameworks like Nunit and I've just figured out that I need to install the adapter to run the tests from the IDE. I'll keep this issue in mind and take a look at what the problem is if nobody else resolved it by then.

**Comment by mrakgr** [comment]
As a preliminary I tested whether this prints out correctly:

    [<Test>]
    let ``Assert String``() =
        let x = "Asd\nQwe"
        AssertionException(x) |> raise

It does in fact. This means that the string is getting unescaped somewhere inside the assert function. I'll figure out when I get to it.

**Comment by mrakgr** [comment]
I tried building NUnit manually, and I can see that the string prints just fine there both in F# (when I reference the manually built dll from Interactive) and C# (when I convert the .NET 4.5 version to a console app.) So that probably means that either the NUnit package that comes with FsUnit is stale, or if the case is that FsUnit fetches the main one directly from Nuget, then the main one on Nuget is stale.

Could you please check that the NUnit that comes with FsCheck is up to date?

**Comment by mrakgr** [comment]
Nevermind, I've verified that it is the main one on Nuget that is causing the issue.

**Comment by sergey-tihon** [comment]
So, here it is https://github.com/nunit/nunit/issues/1575
Will be fixes in `NUnit 3.5`

**Comment by mrakgr** [comment]
Yes, it will be fixed. I think I'll close this issue as it have been investigated. If you want to add structural equality constraints, please feel free to open a separate issue for that.
